blueocean-core-js/src/js/utils/ComponentLink.js

```diff
@@ -0,0 +1,27 @@
+import { PropTypes, Component } from 'react';
+import { SandboxedComponent } from '@jenkins-cd/js-extensions';
+
+export class ComponentLink {
```

Comment:
Base class that wraps the necessary properties for rendering a run action details component link (name, title, and notification data) and the underlying component itself.